File: API/Entities/Booking.cs
namespace API.Entities;

public class Booking
{
    public int Id { get; set; }

    public required string BookingReference { get; set; }

    public int UserId { get; set; }
    public AppUser User { get; set; } = null!;

    public required string BookingMode { get; set; }

    public required string ClientFullName { get; set; }
    public required string ClientEmailAddress { get; set; }
    public required string ClientPhoneNumber { get; set; }
    public required string PreferredContactMethod { get; set; }

    public int TotalDurationMinutes { get; set; }

    public decimal TotalAmount { get; set; }

    public decimal DepositAmount { get; set; }

    public decimal BalanceAmount { get; set; }

    public required string BookingStatus { get; set; }

    public required string PaymentStatus { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? ConfirmedAt { get; set; }

    public DateTime? CancelledAt { get; set; }

    public ICollection<BookingServiceItem> Items { get; set; } = new List<BookingServiceItem>();

    public ICollection<BookingPayment> Payments { get; set; } = new List<BookingPayment>();
}

File: client/src/app/_models/booking.ts
export type BookingMode = 'combined' | 'separate';

export type PreferredContactMethod = 'whatsapp' | 'phone' | 'email' | '';

export interface CreateBookingItem {
  salonServiceId: number;
  salonServiceTypeId: number;
  salonServiceLengthOptionId: number | null;
  appointmentDate: string;
  startTime: string;
  notes?: string | null;
  galleryImageId?: number | null;
  uploadedReferenceImageUrl?: string | null;
}

export interface CreateBooking {
  bookingMode: BookingMode;
  clientFullName: string;
  clientEmailAddress: string;
  clientPhoneNumber: string;
  preferredContactMethod: PreferredContactMethod;
  items: CreateBookingItem[];
}

export interface BookingReviewItem {
  id: number;
  salonServiceId: number;
  serviceName: string;
  serviceTypeName: string;
  lengthName?: string | null;
  appointmentDate: string;
  startTime: string;
  endTime: string;
  durationMinutes: number;
  basePrice: number;
  lengthAddOnPrice: number;
  finalPrice: number;
  notes?: string | null;
  referenceImageUrl?: string | null;
}

export interface BookingReview {
  bookingId: number;
  bookingReference: string;
  bookingMode: BookingMode;
  bookingStatus: string;
  paymentStatus: string;
  clientFullName: string;
  clientEmailAddress: string;
  clientPhoneNumber: string;
  preferredContactMethod: PreferredContactMethod;
  totalDurationMinutes: number;
  totalAmount: number;
  depositAmount: number;
  balanceAmount: number;
  items: BookingReviewItem[];
}